esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+| Property | Count | Description |
+| - | - | - |
+| @database | 1..1 | Name of the database to run the sql statement on. |
+| @MYSQL_... | 0..n | A MySQL option to set on the sql transaction. The full name of the attribute should match a MySQL option enum name e.g. `MYSQL_SET_CHARSET_NAME` |
+| @name | 0..1 | Name of the node inside `@section` where the output is placed. If it does not exist it is created. |
+| @optional | 0..1 | Boolean flag indicating whether script syntax errors are fatal (*false*) or merely generate warnings (*true*). Defaults to *false*. |
+| @password | 0..1 | String of the password to login to the database server. |
+| @resultset-tag | 0..1 | The name of the node holding the resultset generated by executing the sql statement. This node is created in the script context at the location *`section`*`/`*`name`*`/`*`resultset-tag`*|
+| @section | 0..1 | Path to the section of script context where output is placed. If omitted defaults to `temporaries`.|
+| @select | 0..1 | XPath node expression used as root of the source for binding sql parameters. |
+| @server | 0..1 | String value of the database server in the format *ip:port* or *hostname:port*. |
+| @trace | 0..1 | Label used in trace log output messages. If omitted or empty, the element name is used. |
+| @user | 0..1 | String of the user name to login to the database server. |
+| @vault | 0..1 | String of the vault name to retrieve database login info. If not provided, defaults to *esp*. For security, preferred usage is to include user, password and server together. |
+| bind | 0..1 | Node to bind a value to a sql parameter. |
+| bind/@name | 1..1 | String of the name of the sql parameter. |
+| bind/@value | 1..1 | XPath string expression of the value of the parameter. |
+| sql | 1..1 | Required element whose contents are an sql statement to execute. |
+
+TODO: Include examples of usage
+
+### namespace
+```xml
+    <namespace
+        current="Boolen value"
+        optional="Boolean value"
+        prefix="String value"
+        trace="String value"
+        uri="String value"
+    />
+```
+Create or use an XML namespace in the output document.
+
+| Property | Count | Description |
+| - | - | - |
+| @current | 0..1 | Boolean flag indicating whether the identified namespace applies to the output document's current node (*true*) or does not (*false*). Defaults to *true* when `@uri` is empty and *false* otherwise. |
+| @optional | 0..1 | Boolean flag indicating whether script syntax errors are fatal (*false*) or merely generate warnings (*true*). Defaults to *false*. |
+| @prefix | 0..1 | Valid XML namespace prefix string or omitted; must not be empty. Omit to define the default namespace (i.e., the namespace with no prefix). Specify an existing namespace prefix with no `@uri` to refer to the existing namespace. Specify a new prefix with an `@uri` value to define a namespace. |
```

Comment:
This could be empty if the operation turns empty into NULL. Libxml2 wants NULL to identify the default namespace. Should a Jira be opened to make this change?

Reply:
created ticket https://track.hpccsystems.com/browse/HPCC-27936